internal/controller/controllers/controller_event_wrapper.go

```diff
@@ -26,30 +26,23 @@ func NewControllerEventsWrapper(crdEventsHandler CRDEventsHandler, events events
 		events: events, db: db, log: log}
 }
 
-func (c *controllerEventsWrapper) AddEvent(ctx context.Context, clusterID strfmt.UUID, hostID *strfmt.UUID, severity string, msg string, eventTime time.Time, props ...interface{}) {
-	c.events.AddEvent(ctx, clusterID, hostID, severity, msg, eventTime, props)
+func (c *controllerEventsWrapper) V2AddEvent(ctx context.Context, clusterID *strfmt.UUID, hostID *strfmt.UUID, infraEnvID *strfmt.UUID, severity string, msg string, eventTime time.Time, props ...interface{}) {
+	c.events.V2AddEvent(ctx, clusterID, hostID, infraEnvID, severity, msg, eventTime, props)
 
 	if hostID != nil {
-		c.NotifyKubeApiHostEvent(clusterID, *hostID)
+		c.NotifyKubeApiHostEvent(*clusterID, *hostID)
```

Comment:
suggesting to switch to  safe access. not the pointer directly.

Reply:
you can use https://github.com/openshift/assisted-service/blob/f00a9a0da255361695489199579a71c9fe349712/internal/common/conversions.go#L41-L46

Reply:
Done in https://github.com/openshift/assisted-service/pull/3111